reach

```diff
@@ -115,10 +115,103 @@ run_s () {
   sh -ac "$TMP/out.sh" "$0"
 }
 
+package_manager () {
+  if ( which apt > /dev/null 2>&1 ); then
+    export PACKAGE_MANAGER="apt"
+  fi
+
+  if ( which yum > /dev/null 2>&1 ); then
+    export PACKAGE_MANAGER="yum"
+  fi
+}
+
+install_make () {
+  package_manager
+
+  sudo ${PACKAGE_MANAGER} install make
+}
+
+install_curl () {
+  package_manager
+
+  sudo ${PACKAGE_MANAGER} install curl 
+}
+
+timeout_validate () {
+  TIMEOUT=$1
+  STOPPER=$2
+  MESSAGE=$3
+
+  if [ "$STOPPER" -eq "$TIMEOUT" ]; then
+    printf "%s" "$MESSAGE"
+    exit 1;
+  fi
+}
+
+install_docker () {
+  PWD=$(pwd)
+
+  cd / # Depending on `$PWD`, Windows sometimes struggles with paths
+
+  if (which wsl.exe > /dev/null 2>&1); then
+    if (! which winget.exe > /dev/null 2>&1); then
+      printf "Installing Winget Windows Package Manager..."
+      WINGET_VERSION="v1.3.431"
+      WINGET_ARTIFACT_NAME="Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle"
+      WINGET_BUNDLE_URL="https://github.com/microsoft/winget-cli/releases/download/${WINGET_VERSION}/${WINGET_ARTIFACT_NAME}"
+      TEMP_DIR="/mnt/c/tmp"
+      WIN_TEMP_DIR="c:\\tmp"
+      DOCKER_BIN="/mnt/c/Program Files/Docker/Docker/resources/bin/docker"
+      TIMEOUT=180
+      if [ ! -e $TEMP_DIR ]; then
+        mkdir $TEMP_DIR
+      fi
+      sudo wget $WINGET_BUNDLE_URL -O "${TEMP_DIR}/${WINGET_ARTIFACT_NAME}"; powershell.exe "cd ${WIN_TEMP_DIR}; powershell .\\${WINGET_ARTIFACT_NAME}"
+      STOPPER=0
+      while ( ! which winget.exe > /dev/null 2>&1 ); do
+        sleep 1;
+        STOPPER=$(($STOPPER+1))
+        timeout_validate 180 $STOPPER "Winget wasn't installed successfully and timed out!\n"
+      done
+      rm -rf "${TEMP_DIR}/${WINGET_ARTIFACT_NAME}"
+    fi
+    printf "\nInstalling Docker through Winget..."
+    powershell.exe 'Start-Process powershell.exe "winget install docker.dockerdesktop --accept-package-agreements --accept-source-agreements" -Verb runAs'
+    STOPPER=0
+    while [ ! -f "${DOCKER_BIN}" ]; do
+      sleep 120;
+      STOPPER=$(($STOPPER+1))
+      timeout_validate 2 $STOPPER "Docker wasn't installed successfully and timed out!\n"
+    done
+    printf "\nThis installation requires a restart for completion; do you want to restart now? [Y,n]: "
+    read RESPONSE
+    if [ "${RESPONSE}" = "Y" ]; then
+      powershell.exe 'shutdown -r -t 0'
+    fi
+  else
+    printf "Make sure you have Docker installed following these steps:\n"
+    printf "Ubuntu: https://docs.docker.com/engine/install/ubuntu/\n"
+    printf "Debian: https://docs.docker.com/engine/install/debian/\n"
+    printf "CentOS: https://docs.docker.com/engine/install/centos/\n"
+    printf "Fedora: https://docs.docker.com/engine/install/fedora/\n"
+    printf "RedHat: https://docs.docker.com/engine/install/rhel/\n"
+    printf "SLES: https://docs.docker.com/engine/install/SLES/\n"
```

Comment:
```suggestion
    printf "  SLES: https://docs.docker.com/engine/install/SLES/\n"
```

You can ignore this nitpick if you disagree but I really appreciate vertical alignment.